### PR TITLE
More docs

### PR DESCRIPTION
**Pull request by vpipkt**
Note here I moved the function reference to a `pymd` file to enable evaluation of examples.  However not sure how that should work with #173 ... 

Marked as WIP, will continue along with time series, nodata, and pandas/numpy ...  Just shout if anyone wants to go ahead and merge this for any reason.

### CODE REVIEW

**Comment by metasim** [comment]
> Note here I moved the function reference to a pymd file to enable evaluation of examples. However not sure how that should work with #173 ...

If/when we get Sphinx processing of .py files to generate docs, the `reference.pymd` file will likely be obsolete.

**Comment by vpipkt** [comment]
@metasim ready to take a look and merge so we can  get smaller increments of changes from more authors

File: pyrasterframes/src/main/python/docs/nodata-handling.pymd [hunk 1]
```diff
@@ -1,2 +1,158 @@
...
+
+```python ct_from_sen
+spark.read.raster('https://s22s-test-geotiffs.s3.amazonaws.com/luray_snp/B02.tif') \
+    .select(rf_cell_type('proj_raster')).distinct().show()
+```
+
+The function `rf_cell_types` provides a convenient list of basic cell types. Note that this list is not exhaustive. 
+
+```python
+rf_cell_types()
```
Comment (metasim) [inline pyrasterframes/src/main/python/docs/nodata-handling.pymd]: Let's strike this. Maybe replace with the explicit list of "constants" in `CellType`.

File: pyrasterframes/src/main/python/docs/nodata-handling.pymd [hunk 2]
```diff
@@ -1,2 +1,158 @@
...
+unmasked = unmasked.withColumn('nodata', rf_local_equal('scl', lit(0)))
+unmasked = unmasked.withColumn('defect', rf_local_equal('scl', lit(1)))
+unmasked = unmasked.withColumn('cloud8', rf_local_equal('scl', lit(8)))
+unmasked = unmasked.withColumn('cloud9', rf_local_equal('scl', lit(9)))
+unmasked = unmasked.withColumn('cirrus', rf_local_equal('scl', lit(10)))
+
+unmasked = unmasked.withColumn('mask', rf_local_add('nodata', 'defect'))
+unmasked = unmasked.withColumn('mask', rf_local_add('mask', 'cloud8'))
+unmasked = unmasked.withColumn('mask', rf_local_add('mask', 'cloud9'))
+unmasked = unmasked.withColumn('mask', rf_local_add('mask', 'cirrus'))
```
Comment (metasim) [inline pyrasterframes/src/main/python/docs/nodata-handling.pymd]: How about doing chained method invocation... mutability always creeps me out.

File: pyrasterframes/src/main/python/docs/raster-io.md [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
 
 Three types of DataSources will be introduced: 
 
-* Catalog Readers
-    - `aws-pds-l8-catalog`: experimental
-    - `aws-pds-modis-catalog`: experimental
+* @ref:[Catalog Readers](raster-catalogs.md)
+    - `aws-pds-l8-catalog`: built-in catalog over [Landsat on AWS][Landsat]
+    - `aws-pds-modis-catalog`: built-in catalog over [MODIS on AWS][MODIS]
     - `geotrellis-catalog`: for enumerating [GeoTrellis layers][GTLayer]
```
Comment (metasim) [inline pyrasterframes/src/main/python/docs/raster-io.md]: Thank you!!

File: pyrasterframes/src/main/python/docs/raster-read.pymd [hunk 4]
```diff
@@ -38,105 +30,147 @@ parts = rf.select(
...
-print(csv)
+file://  (is this only valid for spark local? or if file is )
+https://
+http:// ?
+s3a://
+s3n:// ?
+hdfs:// ?
+ftp:// ?
+azure ?
+google cloud ?
```
Comment (metasim) [inline pyrasterframes/src/main/python/docs/raster-read.pymd]: We will need to test anything we say we support.

File: pyrasterframes/src/main/python/docs/reference.pymd [hunk 5]
```diff
@@ -44,32 +58,17 @@ Convert an extent to a Geometry. The extent likely comes from @ref:[`st_extent`]
...
-|int32     |
-|float32raw|
-|float32   |
-|float64raw|
-|float64   |
+Print an array of possible cell type names, as below. These names are used in other functions. See @ref:[discussion on NoData](reference.md#masking-and-NoData) for additional details.
 
+```python
+rf_cell_types() 
+```
```
Comment (metasim) [inline pyrasterframes/src/main/python/docs/reference.pymd]: Let's kill this section.

File: pyrasterframes/src/main/python/docs/nodata-handling.pymd (hunk 1 above)
Comment (vpipkt) [reply pyrasterframes/src/main/python/docs/nodata-handling.pymd]: will do

File: pyrasterframes/src/main/python/docs/nodata-handling.pymd (hunk 2 above)
Comment (vpipkt) [reply pyrasterframes/src/main/python/docs/nodata-handling.pymd]: roger doger.

File: pyrasterframes/src/main/python/docs/raster-read.pymd (hunk 4 above)
Comment (vpipkt) [reply pyrasterframes/src/main/python/docs/raster-read.pymd]: perhaps if u can create a unit test I can write it up saying here is what we know works. Let me know

**Pull request merged**